copy/copy.go

```diff
@@ -1166,6 +1166,18 @@ func computeDiffID(stream io.Reader, decompressor compression.DecompressorFunc)
 	return digest.Canonical.FromReader(stream)
 }
 
+type errorAnnotationReader struct {
```

Comment:
Non-blocking: A one-line comment about why this needs to exist would be nice, so that readers 3 years later don’t have to look up the history on GitHub (if it even exists.) I’m sorry, I should have said this earlier.